static/src/javascripts/projects/common/modules/identity/upsell/store/types.js

```diff
@@ -23,6 +23,10 @@ class ConsentWithState {
     setState(hasConsented: boolean) {
         this.hasConsented = hasConsented;
     }
+
+    flipState() {
+        this.hasConsented = !this.hasConsented;
```

Comment:
changeConsentState? flipConsentState?

Reply:
might be a bit redundant? the class is already called `ConsentWithState`. Do you think this might be confusing / did you find this confusing since react uses `state` for a different thing? Happy to change it

Reply:
No all good. Merge away.